### PR TITLE
MAT-5962: Edit Birth Date QI Core Patient

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR

Jira Ticket: [MAT-5962](https://jira.cms.gov/browse/MAT-5962)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/editTestCase/qiCore/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx [hunk 1]
```diff
@@ -141,6 +164,56 @@ const DemographicsSection = ({ canEdit }) => {
           <div className="demographics-container">
             <div className="demographics-row">
               <FormControl tw={"w-2/4"}>
+                <LocalizationProvider dateAdapter={AdapterDayjs}>
```
Comment (adongare) [inline src/components/editTestCase/qiCore/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx]: can we use the date component from the design systems?

File: src/components/editTestCase/qiCore/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx [hunk 2]
```diff
@@ -24,12 +27,24 @@ import {
...
 import {
   ResourceActionType,
   useQiCoreResource,
 } from "../../../../../../util/QiCorePatientProvider";
+import { DatePicker, LocalizationProvider } from "@mui/x-date-pickers";
+import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
+import {
+  birthDateLabelStyle,
+  textFieldStyle,
+} from "./DemographicsSectionStyles";
```
Comment (adongare) [inline src/components/editTestCase/qiCore/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx]: most of these imports seem unused now

File: src/components/editTestCase/qiCore/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx [hunk 3]
```diff
@@ -141,6 +167,19 @@ const DemographicsSection = ({ canEdit }) => {
           <div className="demographics-container">
             <div className="demographics-row">
               <FormControl tw={"w-2/4"}>
+                <DateField
+                  label="Date of Birth"
+                  disabled={!canEdit}
+                  value={patient?.birthDate ? dayjs(patient.birthDate) : null}
+                  handleDateChange={(event: any) => {
+                    const date = new Date(event);
```
Comment (adongare) [inline src/components/editTestCase/qiCore/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx]: creating a date from an "event" seems a bit confusing. I don’t think it is an actual event object. naming it "value" or something other than event would be nice.

File: src/components/editTestCase/qiCore/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx [hunk 4]
```diff
@@ -141,6 +161,19 @@ const DemographicsSection = ({ canEdit }) => {
...
             <div className="demographics-row">
               <FormControl tw={"w-2/4"}>
+                <DateField
+                  label="Date of Birth"
+                  disabled={!canEdit}
+                  value={patient?.birthDate ? dayjs(patient.birthDate) : null}
+                  handleDateChange={(value: any) => {
+                    const date = new Date(value);
+                    const year = date.getFullYear();
+                    const month = String(date.getMonth() + 1).padStart(2, "0"); // Adding 1 because months are zero-indexed
```
Comment (sb-prateekkeerthi) [inline src/components/editTestCase/qiCore/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx]: I think you can remove the commented out code

**Review by sb-prateekkeerthi — CHANGES_REQUESTED** [review]

File: src/components/editTestCase/qiCore/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx (hunk 4 above)
Comment (ethankaplan) [reply src/components/editTestCase/qiCore/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx]: Removed

File: src/components/editTestCase/qiCore/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx [hunk 5]
```diff
@@ -3,9 +3,12 @@ import "twin.macro";
 import "styled-components/macro";
 import { Patient } from "fhir/r4";
 import ElementSection from "../../../../../common/ElementSection";
-
-import { AutoComplete, Select } from "@madie/madie-design-system/dist/react";
-
+import EventIcon from "@mui/icons-material/Event";
```
Comment (sb-prateekkeerthi) [inline src/components/editTestCase/qiCore/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx]: this import can be removed as it is not being used anywhere

File: src/components/editTestCase/qiCore/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx [hunk 6]
```diff
@@ -24,12 +27,18 @@ import {
 } from "./DemographicUtils";
 import "./DemographicsSection.scss";
 import _ from "lodash";
-import { FormControl, MenuItem as MuiMenuItem } from "@mui/material";
+import {
+  FormControl,
+  InputAdornment,
+  InputLabel,
+  MenuItem as MuiMenuItem,
+} from "@mui/material";
```
Comment (sb-prateekkeerthi) [inline src/components/editTestCase/qiCore/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx]: InputAdornment, InputLabel imports can be removed as they are not being used anywhere

**Review by sb-prateekkeerthi — CHANGES_REQUESTED** [review]

**Review by sb-prateekkeerthi — APPROVED** [review]

**Pull request merged**